simd/unit_tests/include/SIMDTesting_Ops.hpp

```diff
@@ -119,4 +119,183 @@ class shift_left {
   }
 };
 
+class cbrt_op {
+ public:
+  template <typename T>
+  auto on_host(T const& a) const {
+#ifdef __INTEL_COMPILER
```

Comment:
Can't you do
```suggestion
#if defined(KOKKOS_COMPILER_INTEL_LLVM) || defined(KOKKOS_COMPILER_INTEL_LLVM)
```
here and in the actual implementation? I would expect that any `oneAPI` compiler would be able to use `svml`.

Reply:
I tried converting relevant lines to `#if defined(KOKKOS_COMPILER_INTEL) || defined(KOKKOS_COMPILER_INTEL_LLVM)`. (I assume you meant this.)
But it seems that the results of `exp` and `log` tests are different between the builds with `icpc`  and `icpx`.
I'm not currently sure whether the values from Kokkos math calls are changing or values from `svml` intrinsic are changing between two builds.
I'll have to look into this a bit more before making this change.

Reply:
I just tested it for `AVX512` using `icpx` and it worked fine. We can look into this in another pull request, though.

Reply:
It does look like the issue that I'm encountering is orthogonal to `svml` usage.

It seems that even with the entire `#ifdef __INTEL_COMPILER` block commented out (i.e. using the fallback `Kokkos::cbrt`/`log`/`exp`), the test fails for `icpx` build while it passes with other compilers including `icpc`. I'm not sure why the behavior differs for `icpx` build specifically , but it looks like there is a bug somewhere that I may need to chase first.

Reply:
Can you provide some more details on the failure? Which `SIMD` type? `Kokkos::Experimental::simd_abi::scalar`?
Maybe, the problem disappears with `-fp-model=precise`?

Reply:
The failure was happening with both `AVX2` and `AVX512`, and the error output is things like this:
```
Expected equality of these values:
  a
    Which is: 0.367879
  b
    Which is: 0.367879
```
which suggests precision errors. So I previously converted unit tests to use gtest assertions that check with floating point precision tolerances, but that didn't fix the problem at the time.

Using `-fp-model=precise` makes the problem disappear. Given this, it was definitely problem with precisions. But since the test doesn't pass without `-fp-model=precise`, I am unsure if it's okay to push through as is or if some workaround is preferrable.